client/main.coffee

```diff
@@ -1275,6 +1290,24 @@ class Board
     @transform.y += @bbox.height/2 * (1/newScale - 1/@transform.scale)
     @transform.scale = newScale
     @retransform()
+  zoomToFit: (min, max) ->
+    # Change the bounds so as to fit the rect from (min, max) on screen
+    width = max.x - min.x
+    height = max.y - min.y
+    midx = 0.5 * (min.x + max.x)
+    midy = 0.5 * (min.y + max.y)
+    @bbox = @svg.getBoundingClientRect()
+    hScale = @bbox.width / width
+    vScale = @bbox.height / height
+    newScale = Math.min hScale, vScale
+    newScale = newScale / 1.05 # Leave a bit of border
```

Comment:
My original plan was to have a margin of 1 grid square.  But 5% seems like a reasonable, perhaps better, choice.